framework/src/actions/MeshOnlyAction.C

```diff
@@ -110,26 +110,32 @@ MeshOnlyAction::act()
           // be outputted to Exodus
           restrict_element_id_names = params.isParamValid("extra_element_ids_to_output");
           if (restrict_element_id_names)
+          {
             element_id_names = params.get<std::vector<std::string>>("extra_element_ids_to_output");
+            // Check which element id names actually are defined on the mesh, remove from
+            // element_id_names if they don't belong
+            for (auto it = element_id_names.begin(); it != element_id_names.end();)
+            {
+              // Erase contents of iterator and return iterator if element integer does not exist
+              if (!output_mesh.has_elem_integer(*it))
+                it = element_id_names.erase(it);
```

Comment:
should we have at least a warning here? we re disobeying a user input

Reply:
Added a warning here